src/overview.md

```diff
@@ -46,45 +46,81 @@ Below is a quick example of a program printing "Pong!" when a ping command comes
 in from a channel:
 
 ```rust
-use futures::StreamExt;
-use twilight_command_parser::{Command, Config as ParserConfig, Parser};
-use twilight_gateway::{Event, Shard};
-use std::{
-    env,
-    error::Error,
-};
+use std::{env, error::Error};
+use tokio::stream::StreamExt;
+use twilight_cache_inmemory::{EventType, InMemoryCache};
+use twilight_gateway::{cluster::{Cluster, ShardScheme}, Event};
+use twilight_http::Client as HttpClient;
+use twilight_model::gateway::GatewayIntents;
 
 #[tokio::main]
-async fn main() -> Result<(), Box<dyn Error>> {
+async fn main() -> Result<(), Box<dyn Error + Send + Sync>> {
     let token = env::var("DISCORD_TOKEN")?;
 
-    // Create a shard, which is the driver for an event loop over incoming
-    // events and sending events.
-    let mut shard = Shard::new(env::var("DISCORD_TOKEN")?);
-    // Retrieve a new asynchronous stream of the events that the shard will
-    // receive.
-    let mut events = shard.events();
-
-    let parser = {
-        let mut config = Config::new();
-        config.command("ping").add();
-        config.add_prefix("!");
-        Parser::new(config)
-    };
-
-    // Start the shard to begin receiving events.
-    shard.start().await?;
-
-    while let Some(event) = events.next().await {
-        match event {
-            Event::MessageCreate(msg) => match parser.parse(&msg.content) {
-                Some(Command { name: "ping", .. }) => println!("Pong!"),
-                _ => {},
-            },
-            // More events here...
-            _ => {},
+    // This is also the default.
+    let scheme = ShardScheme::Auto;
+
+    let cluster = Cluster::builder(&token)
+        .shard_scheme(scheme)
+        // Use intents to only listen to GUILD_MESSAGES events
+        .intents(Some(
+            GatewayIntents::GUILD_MESSAGES | GatewayIntents::DIRECT_MESSAGES,
+        ))
+        .build()
+        .await?;
+
+    // Start up the cluster
+    let cluster_spawn = cluster.clone();
+
+    tokio::spawn(async move {
+        cluster_spawn.up().await;
+    });
+
+    // The http client is seperate from the gateway,
+    // so startup a new one
```

Comment:
"a new one" doesn't really make sense here because the gateway's client is hidden in impl. clarify?